src/sentry/issues/escalating.py

```diff
@@ -0,0 +1,84 @@
+from datetime import datetime, timedelta
+from typing import List, Tuple
+
+from snuba_sdk import (
+    Column,
+    Condition,
+    Direction,
+    Entity,
+    Function,
+    Limit,
+    Offset,
+    Op,
+    OrderBy,
+    Query,
+    Request,
+)
+
+from sentry.models import Group
+from sentry.utils.json import JSONData
+from sentry.utils.snuba import raw_snql_query
+
+# This gets modified in tests to force a smaller limit
+QUERY_LIMIT = 10000  # This is the maximum value for Snuba
+
+
+def query_groups_past_counts(groups: List[Group]) -> JSONData:
+    offset = 0
+    all_results = []
+
+    while True:
+        query = _generate_query(groups, offset)
+        request = Request(dataset="events", app_id="sentry.issues.escalating", query=query)
+        request.validate()
```

Comment:
@evanh should I be validating before making `raw_snql_query` queries? (the original comment conversation was lost when I pushed my latest code)

Reply:
You don't have too, the SnQL code will automatically call `.validate()` before sending the query.